### PR TITLE
Moving editor into a component

### PR DESCRIPTION
**Pull request by grainier**

### CODE REVIEW

**Comment by mohanvive** [comment]
@grainier , please fix above two issues mentioned by @dilini-muthumala and send another PR.

File: modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/DebuggerDataHolder.java [hunk 1]
```diff
@@ -26,6 +27,7 @@
 public class DebuggerDataHolder {
```
Comment (grainier) [inline modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/DebuggerDataHolder.java]: Fixed

File: modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java [hunk 2]
```diff
@@ -40,19 +54,126 @@
...
+            if (in == null) {
+                return null;
+            }
+            File tempFile = File.createTempFile(String.valueOf(in.hashCode()), ".tmp");
+            tempFile.deleteOnExit();
+            FileOutputStream out = new FileOutputStream(tempFile);
+            IOUtils.copy(in, out);
+            return tempFile;
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (grainier) [inline modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java]: Fixed

File: modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java [hunk 3]
```diff
@@ -40,19 +54,126 @@
...
+            // Getting requested stream definitions
+            if (validationRequest.getMissingStreams() != null) {
+                response.setStreams(SourceEditorUtils.getStreamDefinitions(
+                        executionPlanRuntime, validationRequest.getMissingStreams()
+                ));
+            }
+
+            jsonString = new Gson().toJson(response);
+        } catch (Throwable t) {
+            jsonString = new Gson().toJson(new GeneralResponse(Status.ERROR, t.getMessage()));
```
Comment (grainier) [inline modules/components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java]: fixed

**Pull request merged**